service/lib/agama/storage/config_conversions/from_json_conversions/search.rb

```diff
@@ -41,27 +41,20 @@ def convert
           # @see Base#conversions
           # @return [Hash]
           def conversions
+            return convert_string if search_json.is_a?(String)
+
             {
-              name:         convert_name,
-              if_not_found: convert_not_found
+              name:         search_json.dig(:condition, :name),
+              max:          search_json[:max],
+              if_not_found: search_json[:ifNotFound]&.to_sym
             }
           end
 
-          # @return [String, nil]
-          def convert_name
-            return search_json if search_json.is_a?(String)
-
-            search_json.dig(:condition, :name)
-          end
-
-          # @return [Symbol, nil]
-          def convert_not_found
-            return if search_json.is_a?(String)
-
-            value = search_json[:ifNotFound]
-            return unless value
+          # @return [String]
```

Comment:
NP: I think we can follow the same implementation pattern as the rest of conversions:

```ruby
          # @see Base#conversions
          # @return [Hash]
          def conversions
            {
              name:         convert_name,
              max:          search_json[:max],
              if_not_found: convert_not_found
            }
          end

          # @return [String, nil]
          def convert_name
            return if search_json == SEARCH_ANYTHING_STRING
            return search_json if search_json.is_a?(String)

            search_json.dig(:condition, :name)
          end

          # @return [Symbol, nil]
          def convert_not_found
            return :skip if search_json == SEARCH_ANYTHING_STRING
            return if search_json.is_a?(String)
            
            search_json[:ifNotFound]&.to_sym          
          end
```

Reply:
I find that much harder to follow.

We typically handle a hash except an special case in which a string can be used as shortcut.

With the implementation in this PR, that special case is handled at its own method.

With the implementation you suggest, the management is spread over several methods, all of them with its one exception for the string case. And the more attributes we add, the more methods we have with a dedicated `if` at the beginning. I don't see the point.